src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/TypeGVMEntriesNode.cs

```diff
@@ -143,9 +143,7 @@ public IEnumerable<InterfaceGVMEntryInfo> ScanForInterfaceGenericVirtualMethodEn
             {
                 foreach (var method in iface.GetVirtualMethods())
                 {
-                    Debug.Assert(!method.Signature.IsStatic);
-
-                    if (!method.HasInstantiation)
+                    if (method.Signature.IsStatic || !method.HasInstantiation)
```

Comment:
Not sure I understand -- do static interface methods not have slots? And can't they enter the exact diamond problem that's presented below?

Reply:
This matches what we do for non-default methods here:

https://github.com/dotnet/runtime/blob/7db5a577938d83c76f0654635d6ffe1598358097/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/TypeGVMEntriesNode.cs#L106-L107

This code is used to generate a data structure that can be used to resolve generic virtual methods at runtime. We'll stop filtering them here eventually to fix #77070 but a lot more needs to happen to fix that.